### PR TITLE
Convert latin1 to utf8 safe

### PR DESCRIPTION
**Pull request by lemire**
This is an alternative PR to @ronag's PR  (who should be fully credited for the work) at https://github.com/simdutf/simdutf/pull/554

It is basically @ronag's code with 

1. A *proposed* name change... the suffix `_s` is replaced by `_safe`.
2. A simplification: we do not include the new function in the kernels, instead we use @ronag's algorithm on top of existing fast functions.
3. It adds a sanity test.

### CODE REVIEW

**Review by ronag — APPROVED** [review]

**Comment by lemire** [comment]
As soon as the test go green, we will merge and release soon after.

File: README.md
Comment (pauldreik) [inline README.md:866]: nit: indendation

File: README.md
Comment (pauldreik) [inline README.md:869]: may I suggest keeping the name but changing the signature to take a std::span as output? that means the type system conveys the meaning. std span is C++20 which may be a no go.

File: include/simdutf/implementation.h
Comment (pauldreik) [inline include/simdutf/implementation.h:229]: nit x2: indentation

File: README.md
Comment (lemire) [reply README.md:869]: > std span is C++20 which may be a no go.

We support C++11. We could add `std::span` support, but it would have to be optional (so, on top of the existing API). Note that this applies to our whole API. I will add an issue.

Comment (lemire) [reply README.md:869]: Done : https://github.com/simdutf/simdutf/issues/557

File: include/simdutf/implementation.h
Comment (lemire) [reply include/simdutf/implementation.h]: Thanks. Fixed.

File: README.md
Comment (lemire) [reply README.md]: Thanks. Fixed.

**Comment by pauldreik** [comment]
I added a fuzzer for this, it found nothing. I will make a PR later. In the meanwhile, it can be reviewed here: https://github.com/pauldreik/simdutf/commit/7bf34b6f5bc3e757923b000ad952cebaab10043e

**Comment by lemire** [comment]
@pauldreik Fantastic!!!

**Comment by lemire** [comment]
I am merging this, but we will wait for the fuzzer before releasing.

**Comment by lemire** [comment]
@ronag Note that the PR is merged with your credit:

Screenshot 2024-09-03 at 4 03 12 PM

**Comment by pauldreik** [comment]
there seems to be potential for improvement or optimization based on the coverage from the fuzzer (or possibly, the fuzzer is not able to generate all possible interesting input).
this is from https://storage.googleapis.com/oss-fuzz-coverage/simdutf/reports/20240905/linux/src/simdutf/src/scalar/latin1_to_utf8/latin1_to_utf8.h.html

bild

ping @lemire

**Comment by ronag** [comment]
Not sure what that means? Does it mean that it never gets there? It's a very edge casy condition that needs very specific state to occur.

**Comment by pauldreik** [comment]
> Not sure what that means? Does it mean that it never gets there?
yes. the light blue (and red) numbers are the hit counts from the fuzz corpus.


>  It's a very edge casy condition that needs a very specific state to occur.
Can you provide me with an example of input data that gets there?

**Comment by lemire** [comment]
@pauldreik Hmmmm.... What aboout the single byte `0x80` with `utf8_len` equal to 1?

**Pull request merged**
